pkg/reconciler/labeler/labeler_test.go

```diff
@@ -74,6 +74,24 @@ func TestReconcile(t *testing.T) {
 			Eventf(corev1.EventTypeNormal, "FinalizerUpdate", "Updated %q finalizers", "first-reconcile"),
 		},
 		Key: "default/first-reconcile",
+	}, {
+		Name: "label pinned revision",
+		Objects: []runtime.Object{
+			simpleRoute("default", "pinned-revision", "the-revision"),
+			simpleConfig("default", "red-herring", WithLatestCreated("red-herring-ecoge")),
+			rev("default", "red-herring"),
+			rev("default", "red-herring", WithRevName("red-herring-ecoge")),
+			rev("default", "red-herring", WithRevName("the-revision")),
+		},
+		WantPatches: []clientgotesting.PatchActionImpl{
+			patchAddFinalizerAction("default", "pinned-revision"),
+			patchAddLabel("default", "the-revision",
+				"serving.knative.dev/route", "pinned-revision"),
+		},
+		WantEvents: []string{
+			Eventf(corev1.EventTypeNormal, "FinalizerUpdate", "Updated %q finalizers", "pinned-revision"),
+		},
+		Key: "default/pinned-revision",
```

Comment:
Hey @mattmoor, in this test scenario, should the configuration get a label as well?

Reply:
I think it should.

Reply:
From slack:
> We want the configuration labeled whenever it or one of its revisions are in a route